src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMNavigator.java

```diff
@@ -107,8 +109,12 @@ public String getApiUri() {
             listener.getLogger().format("No scan credentials, skipping%n");
             return;
         }
-        listener.getLogger().format("Connecting to GitHub using %s%n", CredentialsNameProvider.name(credentials));
         GitHub github = Connector.connect(apiUri, credentials);
+        if (!github.isCredentialValid()) {
+            listener.getLogger().format("Invalid scan credentials, skipping%n");
+            return;
```

Comment:
This implies that credentials `- none -` is not valid the _Scan Credentials_ dropdown list. Do we want that?